src/vector_index.c

```diff
@@ -238,7 +242,49 @@ int VecSim_RdbLoad(RedisModuleIO *rdb, VecSimParams *vecsimParams) {
     vecsimParams->hnswParams.efRuntime = LoadUnsigned_IOError(rdb, goto fail);
     break;
   }
-  return REDISMODULE_OK;
+  // Checking if the loaded parameters fits the current server limits.
+  rv = VecSimIndex_validate_params(ctx, vecsimParams, &status);
+  if (REDISMODULE_OK != rv) {
+    size_t old_block_size = 0;
+    size_t old_initial_cap = 0;
+    RedisModule_LogIOError(rdb, REDISMODULE_LOGLEVEL_WARNING, "ERROR: %s", QueryError_GetError(&status));
+    // We change the initial size to 0 and block size to default and try again.
+    // setting block size to 0 will later set it to default. 
+    switch (vecsimParams->algo) {
+      case VecSimAlgo_BF:
+        old_block_size = vecsimParams->bfParams.blockSize;
+        old_initial_cap = vecsimParams->bfParams.initialCapacity;
+        vecsimParams->bfParams.blockSize = 0;
+        vecsimParams->bfParams.initialCapacity = 0;
+        break;
+      case VecSimAlgo_HNSWLIB:
+        old_block_size = vecsimParams->hnswParams.blockSize;
+        old_initial_cap = vecsimParams->hnswParams.initialCapacity;
+        vecsimParams->hnswParams.blockSize = 0;
+        vecsimParams->hnswParams.initialCapacity = 0;
+        break;
+    }
+    // If we changed the initial capacity, log the change
+    if (old_initial_cap) {
+      RedisModule_LogIOError(rdb, REDISMODULE_LOGLEVEL_WARNING, "WARNING: changing initial capacity from %zu to 0", old_initial_cap);
+    }
+    QueryError_ClearError(&status);
+    // We don't know yet what the block size will be (default value can be effected by memory limit),
+    // so we first validating the new parameters.
+    rv = VecSimIndex_validate_params(ctx, vecsimParams, &status);
+    // Now default block size is set. we can log this change now.
+    if (VecSimAlgo_HNSWLIB == vecsimParams->algo && vecsimParams->hnswParams.blockSize != old_block_size) {
+      RedisModule_LogIOError(rdb, REDISMODULE_LOGLEVEL_WARNING, "WARNING: changing block size from %zu to %zu", old_block_size, vecsimParams->hnswParams.blockSize);
```

Comment:
can you test this?

Reply:
Not right now because the HNSW block size can only be determent automatically.